File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.kt
/**
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */

class Solution {
    fun levelOrder(root: TreeNode?): List<List<Int>> {
        val res = mutableListOf<MutableList<Int>>()
        if(root == null)
            return res
        var root = root
        val que = ArrayDeque<TreeNode>()
        que.add(root)

        while(!que.isEmpty()){
            val temp = mutableListOf<Int>()
            for (i in 0 .. que.size -1)
            {
                val root = que.removeFirst()
                temp.add(root.`val`)
                if(root.left != null) que.add(root.left)
                if(root.right != null) que.add(root.right)
            }
            res.add(temp)
        }
        return res
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-2.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        if n == 2:
            return 2
        h1, h0= 2, 1
        for i in range(2, n):
            temp = h1
            h1 = h1 + h0
            h0 = temp

        return h1